model/task_lifecycle.go

```diff
@@ -185,7 +185,7 @@ func TryResetTask(taskId, user, origin string, detail *apimodels.TaskEndDetail)
 	}
 
 	// only allow re-execution for failed or successful tasks
-	if !t.IsFinished() {
+	if !t.IsFinished() && !t.IsBlockedDisplayTask() {
```

Comment:
Is this the right place to put this? `TryResetTask` is called in a lot of automated cases, not just manual restarts. Since `FindAllTasksFromVersionWithDependencies` is potentially expensive—for versions with many display tasks, each query will return many tasks, and there may be many queries—should we check this earlier, like in the routes to reset a display task?

Reply:
I had considered this but I believe this is fine. The biggest thing is that it no-ops immediately for execution tasks, and only does queries when we would be restarting display tasks which should be rare